### PR TITLE
Geometry Clang Check modernize-loop-convert

### PR DESCRIPTION
**Pull request by ianna**
Geometry part of @davidlange6's PR #19649

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
@kpedro88 - please have a look

**Comment by davidlange6** [comment]
merge

**Comment by kpedro88** [comment]
@davidlange6 I was 3/4 through a review of this...

File: Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLayerBuilder.cc [hunk 1]
```diff
@@ -71,21 +71,21 @@ void CmsTrackerLayerBuilder::sortNS(DDFilteredView& fv, GeometricDet* det){
       }
     }
 
-    for(GeometricDet::GeometricDetContainer::iterator i=neg.begin();i!=neg.end();i++){
-      double rPos = (*i)->translation().Rho();
+    for(auto & i : neg){
```
Comment (kpedro88) [inline Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLayerBuilder.cc]: `const`?

File: Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLayerBuilder.cc [hunk 2]
```diff
@@ -71,21 +71,21 @@ void CmsTrackerLayerBuilder::sortNS(DDFilteredView& fv, GeometricDet* det){
...
+	extneg.push_back(i);
       }else{
-	intneg.push_back(*i);
+	intneg.push_back(i);
       }
     }
 
-    for(GeometricDet::GeometricDetContainer::iterator i=pos.begin();i!=pos.end();i++){
-      double rPos = (*i)->translation().Rho();
+    for(auto & po : pos){
```
Comment (kpedro88) [inline Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLayerBuilder.cc]: `const`?

File: Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLevelBuilder.h [hunk 3]
```diff
@@ -117,16 +117,16 @@ class CmsTrackerLevelBuilder : public CmsTrackerAbstractConstruction {
       float phi = 0.;
       bool sum = true;
       
-      for(unsigned int i=0;i<comp.size();i++){
-	if(fabs(comp[i]->phi())>pi/2.) {
+      for(auto & i : comp){
```
Comment (kpedro88) [inline Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLevelBuilder.h]: remove `&`?

File: Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLevelBuilder.h [hunk 4]
```diff
@@ -117,16 +117,16 @@ class CmsTrackerLevelBuilder : public CmsTrackerAbstractConstruction {
...
+	if(fabs(i->phi())>pi/2.) {
 	  sum = false;
 	  break;
 	}
       }
       
       if(sum){
-	for(unsigned int i=0;i<comp.size();i++){
-	  phi+= comp[i]->phi();
+	for(auto & i : comp){
```
Comment (kpedro88) [inline Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLevelBuilder.h]: remove `&`?

File: Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLevelBuilder.h [hunk 5]
```diff
@@ -135,9 +135,9 @@ class CmsTrackerLevelBuilder : public CmsTrackerAbstractConstruction {
 	return temp;
 	
       }else{
-	for(unsigned int i=0;i<comp.size();i++){
-	  double phi1 = comp[i]->phi() >= 0 ? comp[i]->phi(): 
-	    comp[i]->translation().phi()+2*pi; 
+	for(auto & i : comp){
```
Comment (kpedro88) [inline Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLevelBuilder.h]: remove `&`?

File: Geometry/TrackerNumberingBuilder/src/TrackerMapDDDtoID.cc [hunk 6]
```diff
@@ -30,11 +30,11 @@ void TrackerMapDDDtoID::buildAllStep2(const GeometricDet* theTracker){
   // Also build a map! (for slower access)
   //
 
-  for (unsigned int j=0; j<allDetectors.size(); j++){
+  for (auto & allDetector : allDetectors){
```
Comment (kpedro88) [inline Geometry/TrackerNumberingBuilder/src/TrackerMapDDDtoID.cc]: `const`?

File: Geometry/TrackerNumberingBuilder/test/GeometricDetAnalyzer.cc [hunk 7]
```diff
@@ -101,8 +101,8 @@ GeometricDetAnalyzer::analyze( const edm::Event& iEvent, const edm::EventSetup&
    edm::LogInfo("GeometricDetAnalyzer")<< " Top node is  "<< pDD.product();   
    edm::LogInfo("GeometricDetAnalyzer")<< " And Contains  Daughters: "<< pDD->deepComponents().size();   
    std::vector<const GeometricDet*> det = pDD->deepComponents();   
-   for(std::vector<const GeometricDet*>::iterator it = det.begin();it!=det.end();it++){
-     DDRotationMatrix res = (*it)->rotation();
+   for(auto & it : det){
```
Comment (kpedro88) [inline Geometry/TrackerNumberingBuilder/test/GeometricDetAnalyzer.cc]: remove `&`?

File: Geometry/TrackerNumberingBuilder/test/ModuleNumbering.cc [hunk 8]
```diff
@@ -206,8 +206,8 @@ ModuleNumbering::analyze( const edm::Event& iEvent, const edm::EventSetup& iSetu
   std::vector<const GeometricDet*> modules =  (*rDD).deepComponents();
   std::map< uint32_t , const GeometricDet* > mapDetIdToGeometricDet;
   
-  for(unsigned int i=0; i<modules.size();i++){  
-    mapDetIdToGeometricDet[modules[i]->geographicalID().rawId()] = modules[i];
+  for(auto & module : modules){
```
Comment (kpedro88) [inline Geometry/TrackerNumberingBuilder/test/ModuleNumbering.cc]: `const`?

File: Geometry/VeryForwardGeometryBuilder/plugins/GeometryInfoModule.cc [hunk 9]
```diff
@@ -216,9 +216,9 @@ void GeometryInfoModule::PrintGeometry(const TotemRPGeometry &geometry, const ed
     printf("RPId |                center                |      U direction    |      V direction    |\n");
     printf("     |   x (mm)   |   y (mm)   |   z  (m)   |    dx    |    dy    |    dx    |    dy    |\n");
     
-    for (map<unsigned int, MeanRPData>::iterator it = data.begin(); it != data.end(); ++it)
+    for (auto & it : data)
```
Comment (kpedro88) [inline Geometry/VeryForwardGeometryBuilder/plugins/GeometryInfoModule.cc]: `const`?

File: Geometry/VeryForwardGeometryBuilder/plugins/TotemRPIncludeAlignments.cc [hunk 10]
```diff
@@ -229,12 +229,12 @@ void TotemRPIncludeAlignments::setIntervalFor(const edm::eventsetup::EventSetupR
   bool next_exists = false;
   TimeValue_t t = iosv.time().value(), next_start = TimeValidityInterval::EndOfTime();
 
-  for (RPAlignmentCorrectionsDataSequence::iterator it = seq->begin(); it != seq->end(); ++it)
+  for (auto & it : *seq)
```
Comment (kpedro88) [inline Geometry/VeryForwardGeometryBuilder/plugins/TotemRPIncludeAlignments.cc]: `const`?

File: Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc [hunk 11]
```diff
@@ -149,8 +148,8 @@ DetGeomDesc::ConstContainer DetGeomDesc::deepComponents() const
 
 void DetGeomDesc::addComponents(Container cont)
 {
-	for( Container::iterator ig = cont.begin(); ig != cont.end();ig++) {
-		_container.push_back(*ig);
+	for(auto & ig : cont) {
```
Comment (kpedro88) [inline Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc]: remove `&`?

**Review by kpedro88 — COMMENTED** [review]
I noticed a number of places where const and reference usage could be improved. There were also a few instances where loops could be converted to range-based but weren't. In general, I think we should be consistent in using copies for primitive types (and pointers), and references for everything else (`const` when possible).

File: Geometry/CaloEventSetup/plugins/CaloTowerConstituentsMapBuilder.cc [hunk 12]
```diff
@@ -130,17 +130,16 @@ void CaloTowerConstituentsMapBuilder::assignEEtoHE(const CaloGeometry* geometry,
   //get list of EE detids
   const std::vector<DetId>& vec(geomEE->getValidDetIds());
   //loop over EE detids
-  for(std::vector<DetId>::const_iterator detId_itr =vec.begin();
-      detId_itr != vec.end(); detId_itr++){
+  for(auto detId_itr : vec){
```
Comment (makortel) [inline Geometry/CaloEventSetup/plugins/CaloTowerConstituentsMapBuilder.cc]: On the other hand DetId is just a `uint32_t`.

Comment (kpedro88) [inline Geometry/CaloEventSetup/plugins/CaloTowerConstituentsMapBuilder.cc]: Is copying a class whose only member is a primitive just as cheap as copying a primitive itself? I admit, I'm not sure.

Comment (Dr15Jones) [inline Geometry/CaloEventSetup/plugins/CaloTowerConstituentsMapBuilder.cc]: It should be as long as the class used the compiler generated default constructors.

Comment (kpedro88) [inline Geometry/CaloEventSetup/plugins/CaloTowerConstituentsMapBuilder.cc]: It looks like `DetId` has user-defined default constructors:
https://github.com/cms-sw/cmssw/blob/master/DataFormats/DetId/interface/DetId.h#L26-L32
as do some of the derived types, e.g.:
https://github.com/cms-sw/cmssw/blob/master/DataFormats/HcalDetId/interface/HcalDetId.h#L33-L42
As I would expect based on that, both `DetId` and `HcalDetId` fail `std::is_pod`.

**Pull request merged**